src/mod_stream_management.erl

```diff
@@ -115,10 +163,84 @@ get_resume_timeout(Default) ->
 set_resume_timeout(ResumeTimeout) ->
     set_module_opt(?MYNAME, ?MODULE, resume_timeout, ResumeTimeout).
 
+
+-spec get_gc_repeat_after(pos_integer()) -> pos_integer().
+get_gc_repeat_after(Default) ->
+    MaybeModOpts = gen_mod:get_module_opt(?MYNAME, ?MODULE, stale_h, undefined),
+    case MaybeModOpts of
+        {true, GCOpts} ->
+            proplists:get_value(gc_repeat_after, GCOpts, Default);
+        _ -> Default
+    end.
+
+-spec set_gc_repeat_after(pos_integer()) -> boolean().
+set_gc_repeat_after(ResumeTimeout) ->
+    MaybeModOpts = gen_mod:get_module_opt(?MYNAME, ?MODULE, stale_h, undefined),
+    case MaybeModOpts of
+        {true, GCOpts} ->
+            NewGCOpts = lists:keystore(gc_repeat_after, 1, GCOpts,
+                                       {gc_repeat_after, ResumeTimeout}),
+            NewStaleOpts = {true, NewGCOpts},
+            set_module_opt(?MYNAME, ?MODULE, stale_h, NewStaleOpts);
+        _ -> false
+    end.
+
+-spec get_gc_geriatric(pos_integer()) -> pos_integer().
+get_gc_geriatric(Default) ->
+    MaybeModOpts = gen_mod:get_module_opt(?MYNAME, ?MODULE, stale_h, undefined),
+    case MaybeModOpts of
+        {true, GCOpts} ->
+            proplists:get_value(gc_geriatric, GCOpts, Default);
+        _ -> Default
+    end.
+
+-spec set_gc_geriatric(pos_integer()) -> boolean().
+set_gc_geriatric(ResumeTimeout) ->
```

Comment:
And this is the exact reason for using flat options - you don't need extra helpers.

Btw, can we name it gc_ttl?

Anyway, you don't need three copies of the same function, parametrize it and make something like:

set_stale_h_option(...., StaleHOptName, StaleHOptValue)